### ISSUE TITLE
Installation issue: mesa

### ISSUE THREAD

**Issue by steo85it** [LLNL/spack#26595]
### Steps to reproduce the issue

```console
spack install mesa
```

### Information on your system

* **Spack:** 0.16.3-4766-449a5832c8
* **Python:** 3.6.8
* **Platform:** linux-centos7-skylake
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/7308314/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/7308316/spack-build-env.txt)

Thanks for your help with this!
@chuckatkins  @v-dobrev

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by bernhardkaindl** [comment]
@steo85it : The actual build error is not found in your build log AFAICS (it is truncated, it was likely too long).

But, build with LLVM 13, like your build failed for me too.

My problem was fixed by using LLVM 12 instead: #26627
(It's quite plausible that Mesa's Software Rasterizer does need updates for LLVM 13)

Without any change to your spack tree, you can confirm if the fix works for you too with:
```py
spack install mesa ^llvm@12.0.1
```

**Comment by steo85it** [comment]
Ops, my bad, sorry!! Your suggestion works, anyway: I was able to install mesa with llvm 12 (maybe it would be good to modify the dependency requirements?). Thanks!

**Comment by bernhardkaindl** [comment]
> Ops, my bad, sorry!! Your suggestion works, anyway: I was able to install mesa with llvm 12 (maybe it would be good to modify the dependency requirements?). Thanks!

I did now, @tldahlgren and @prckent  (problem fixed: Mesa does not build with LLVM@13, @13 is for now excluded from building Mesa). 

@steo85it  No not your bad, but when adding new a new major version of a compiler, we have to expect that intimate users of it could be affected until we test and are able to correct for it. 

It's the 2nd build issue related to LLVM@13 that I fixed btw, the first one was #26584.

And the ext one to fix is #26633, so we playing catch up now for not putting in all the checks first ;)
